### ISSUE TITLE
"Error -4081 ECANCELED operation canceled" on page refreshed

### ISSUE THREAD

**Issue by paolotagliaferri** [aspnet/KestrelHttpServer#1855]
Hello everybody!

I'm new to ASP.NET Core and not sure that what I found is the expected behavior or not.

I recently followed [this tutorial ](https://channel9.msdn.com/Events/Visual-Studio/Visual-Studio-2017-Launch/WEB-103) and created my first project. Since I experienced slow performance and sometimes even timeout exceptions when deploying the solution on the local IIS, I started using the command prompt to launch the web application.

When I run it in development mode (ASPNETCORE_ENVIRONMENT=Development), every time I refresh any page of the web app I get a log like this:

```
Request starting HTTP/1.1 GET http://localhost:5000/__webpack_hmr
info: Microsoft.AspNetCore.Server.Kestrel[14]
      Connection id "0HL52ENICB5GT" communication error.
Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4081 ECANCELED operation canceled
```

The page loads correctly on the browser, but I wonder if this is a symptom of something wrong happening on the server due to bad code or configuration.

In case you want to reproduce the issue you can get the solution on [this repository](https://github.com/CRANK211/vs17-ng2-dnc) and use the _completed-sample_ folder. Then run it with:
```
dotnet restore
npm install
dotnet build
set ASPNETCORE_ENVIRONMENT=Development
dotnet run
```

A brief recap of my environment:

- Windows 7
- Visual Studio 2017
- Asp.Net Core 1.1
- Node.JS 7.10 and Angular 2

Hope it's helpful! :smiley:

**Comment by halter73** [comment]
I was able to repro your issue. It seems like this is the same issue reported in #1103. The root problem seems to be the RequestAborted token [used here](https://github.com/aspnet/JavaScriptServices/blob/cdbbbcc473c7da06f240253d6c60c8e3518f5be6/src/Microsoft.AspNetCore.SpaServices/Webpack/ConditionalProxyMiddleware.cs#L101) doesn't fire immediately when the client disconnects, but only if there is a connection error.

Well... since the client disconnects and the RequestAborted doesn't fire, the ConditionalProxyMiddleware keeps on writing to the http://localhost:5000/__webpack_hmr response which eventually causes a connection error to be observed and logged because the client is disconnected.

To fix #1103, we considered creating a new ClientDisconnected token, but instead we decided to always fire the RequestAborted token immediately even when the client disconnects gracefully.

The change to the RequestAborted token (#1218) is in our 2.0.0 releases which should fix the communication error logs since ConditionalProxyMiddleware will stop writing as soon as the client disconnects. In the meantime, it is safe to ignore these logs. This is partially why this is only logged as "info" and not something like "warning" or "error".
